### PR TITLE
Add ls capabilities for partial parser service

### PR DESCRIPTION
**Pull request by madushajg**
## Purpose
$Subject
Related to: https://github.com/ballerina-platform/ballerina-lang/issues/36448

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: misc/ls-extensions/modules/partial-parser/src/main/java/io.ballerina.parsers/PartialParserClientCapabilities.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.ballerina.parsers;
+
+import org.ballerinalang.langserver.commons.registration.BallerinaClientCapability;
+
+/**
+ * Client capabilities for the jsonToRecord service.
```
Comment (prabushi) [inline misc/ls-extensions/modules/partial-parser/src/main/java/io.ballerina.parsers/PartialParserClientCapabilities.java:21]: Update comments.

File: misc/ls-extensions/modules/partial-parser/src/main/java/io.ballerina.parsers/PartialParserClientCapabilities.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ *
+ * @since 2.0.0
```
Comment (prabushi) [inline misc/ls-extensions/modules/partial-parser/src/main/java/io.ballerina.parsers/PartialParserClientCapabilities.java:23]: @IMS94 Should the version be 2201.1.1?

Comment (IMS94) [reply misc/ls-extensions/modules/partial-parser/src/main/java/io.ballerina.parsers/PartialParserClientCapabilities.java:23]: Yes

File: misc/ls-extensions/modules/partial-parser/src/main/java/io.ballerina.parsers/PartialParserClientCapabilitySetter.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ */
+package io.ballerina.parsers;
+
+import org.ballerinalang.annotation.JavaSPIService;
+import org.ballerinalang.langserver.commons.registration.BallerinaClientCapabilitySetter;
+
+/**
+ * Client Capability setter for the {@link PartialParserService}.
+ *
+ * @since 2.0.0
```
Comment (IMS94) [inline misc/ls-extensions/modules/partial-parser/src/main/java/io.ballerina.parsers/PartialParserClientCapabilitySetter.java:24]: Please change this to 2201.1.1 as well.

**Review by IMS94 — CHANGES_REQUESTED** [review]
LGTM! Please update the version in javadocs

File: misc/ls-extensions/modules/partial-parser/src/main/java/io.ballerina.parsers/PartialParserClientCapabilitySetter.java (hunk 3 above)
Comment (madushajg) [reply misc/ls-extensions/modules/partial-parser/src/main/java/io.ballerina.parsers/PartialParserClientCapabilitySetter.java]: Fixed with 9bf9ee11894fa6e30e5c97e65a867c2f941b3710

File: misc/ls-extensions/modules/partial-parser/src/main/java/io.ballerina.parsers/PartialParserClientCapabilities.java (hunk 2 above)
Comment (madushajg) [reply misc/ls-extensions/modules/partial-parser/src/main/java/io.ballerina.parsers/PartialParserClientCapabilities.java]: Fixed with 9bf9ee11894fa6e30e5c97e65a867c2f941b3710

**Review by IMS94 — APPROVED** [review]

**Pull request merged**